crates/matrix-sdk-indexeddb/Cargo.toml

```diff
@@ -14,8 +14,9 @@ default-target = "wasm32-unknown-unknown"
 rustdoc-args = ["--cfg", "docsrs"]
 
 [features]
-default = ["e2e-encryption"]
-e2e-encryption = ["matrix-sdk-base/e2e-encryption", "dep:matrix-sdk-crypto"]
+default = ["e2e-encryption", "state-store"]
+state-store = ["dep:matrix-sdk-base"]
+e2e-encryption = ["matrix-sdk-base?/e2e-encryption", "dep:matrix-sdk-crypto"]
```

Comment:
The dependency on `matrix-sdk-base/e2e-encryption` is needed here to make `make_store_config` work (in other words, we only need `matrix-sdk-base/e2e-encryption` if *both* `state-store` and `e2e-encryption` are enabled, and afaik this is the least bad way to achieve that in Cargo.toml.

That said: I think this is defeating the point of this PR, which was to avoid having to compile `matrix-sdk-base` for `matrix-sdk-crypto-wasm`, which does not use the state store. In a future PR I think I will investigate moving `make_store_config` to `matrix-sdk`, which will make everything simpler.